crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -897,7 +921,7 @@ pub mod tests {
         (git_command_provider, tempdir_handle)
     }
 
-    async fn commit_file(repo: &GitCommandProvider, filename: &str) {
```

Comment:
suggestion: make inherent method

besides having both async and sync methods, also mixing free functions and inherent methods is making this module become messier than it should be.
Also making test methods public is weird, as they wont work in non-test code.

Reply:
Discussed live